services/user.js

```diff
@@ -227,13 +236,19 @@ class UserService {
 
     async listUsers(params, context) {
         verifySession(context)
-            .verifyInitialized()
-            .verifyPermission(USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
+            .verifyInitialized();
+        await this.#isPermitted(context?.userInfo, USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
+        const isFederalLeadOnly = await this.#isManageUserScope(context?.userInfo, ROLES.FEDERAL_LEAD);
 
-        const result = await this.userCollection.aggregate([{
-            "$match": {}
-        },]);
+        if (context?.userInfo?.role === ROLES.FEDERAL_LEAD && !isFederalLeadOnly) {
+            return [];
+        }
 
+        const result = await this.userCollection.aggregate([{
+            "$match": {
+                ...(isFederalLeadOnly ? { role: ROLES.FEDERAL_LEAD } : {})
```

Comment:
Same as above, if scope is "all" then no filters. If scope is "role", then use the scope value returned by authorization function to create the filter.

Reply:
Yes, the filter is applied for the role scope.